anillo/http/request.py

```diff
@@ -1,24 +1,16 @@
-from werkzeug.wrappers import BaseRequest
-from werkzeug.wrappers import AcceptMixin
-from werkzeug.wrappers import ETagRequestMixin
-from werkzeug.wrappers import UserAgentMixin
-from werkzeug.wrappers import AuthorizationMixin
-from werkzeug.wrappers import CommonRequestDescriptorsMixin
-from werkzeug.wrappers import StreamOnlyMixin
+from werkzeug.datastructures import EnvironHeaders
+from werkzeug._compat import wsgi_get_bytes
+from werkzeug.wsgi import get_current_url, get_input_stream
 
-
-class Request(BaseRequest, AcceptMixin, ETagRequestMixin,
-              UserAgentMixin, AuthorizationMixin,
-              CommonRequestDescriptorsMixin):
-    """Full featured request object implementing the following mixins:
-
-    - :class:`AcceptMixin` for accept header parsing
-    - :class:`ETagRequestMixin` for etag and cache control handling
-    - :class:`UserAgentMixin` for user agent introspection
-    - :class:`AuthorizationMixin` for http auth handling
-    - :class:`CommonRequestDescriptorsMixin` for common headers
-    """
-
-
-class PlainRequest(StreamOnlyMixin, Request):
-    """A request object without special form parsing capabilities."""
+def environ_to_request(environ):
+    return {
+        "server-port": environ.get('SERVER_PORT', None),
+        "server-name": environ.get('SERVER_NAME', None),
+        "remote-addr": environ.get('REMOTE_ADDR', None),
+        "uri": get_current_url(environ),
+        "query-string": wsgi_get_bytes(environ.get('QUERY_STRING', '')),
+        "scheme": environ.get('SERVER_PROTOCOL', None),
+        "request-method": environ.get('REQUEST_METHOD', "").upper(),
+        "headers": dict(EnvironHeaders(environ)),
```

Comment:
Maybe it make sense keep EnvironHeaders instance, because it has case insensitive access to it keys, and maybe other useful things...